pkg/idl/listtype_test.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package idl_test
+
+// This example shows how you can use the listType map attribute and how
+// to specify a key that will be used to identify specific elements of
+// the list. The listMapKey attribute is used to specify that Name will
+// be used as the key of the map.
+func ExampleListType_map() {
+	type SomeAPI struct {
+		// +listType=map
+		// +listMapKey=name
+		elements []struct {
```

Comment:
This is confusing because the type is inlined, so it isn't clear whether it is on the field or the type.  Pull the struct out into its own type definition to make this clear.

Reply:
Please just add a section explaining this. :)

Reply:
Yeah, obviously :-) I've just started with the `listType` attribute so far!